crates/amaru/src/stages/consensus/fetch_block.rs

```diff
@@ -47,7 +47,13 @@ impl BlockFetchStage {
 
     async fn handle_event(&mut self, event: ValidateHeaderEvent) -> Result<(), WorkerError> {
         match event {
-            ValidateHeaderEvent::Validated { peer, point, span } => {
+            ValidateHeaderEvent::Validated { peer, header, span } => {
+                let point = match header.header_body.prev_hash {
+                    None => Point::Origin,
+                    Some(_) => {
+                        Point::Specific(header.header_body.slot, header.body_signature.to_vec())
```

Comment:
Sorry, this is wrong, it should be  `header.hash()`  that's passed to `Specific` not the signature